webapp/graphite/remote_storage.py

```diff
@@ -1,44 +1,49 @@
 import time
 import httplib
+import urllib3
+from Queue import Queue
 from urllib import urlencode
-from threading import Lock
+from threading import Lock, current_thread
 from django.conf import settings
 from django.core.cache import cache
+from graphite.intervals import Interval, IntervalSet
 from graphite.node import LeafNode, BranchNode
-from graphite.readers import FetchInProgress
 from graphite.logger import log
 from graphite.util import unpickle
+from graphite.readers import FetchInProgress
 from graphite.render.hashing import compactHash
-from graphite.intervals import Interval, IntervalSet
+from graphite.worker_pool.pool import get_pool
+
+http = urllib3.PoolManager(num_pools=10, maxsize=5)
 
 def connector_class_selector(https_support=False):
     return httplib.HTTPSConnection if https_support else httplib.HTTPConnection
 
+
 class RemoteStore(object):
   lastFailure = 0.0
```

Comment:
@DanCech I'm wondering why they've had this as a class variable. This might be a bug, because it will mark all remote storages of this type as failed if one failed, or maybe that's the expected behavior but i don't understand why. I think I should just remove this property because it's not used anymore.